src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -206,15 +206,19 @@ def kubernetes_exception_handler(ex, fault_type, summary, error_message='Error o
         status_code = ex.status
         if status_code == 403:
             logger.warning(message_for_unauthorized_request)
-        if status_code == 404:
+        elif status_code == 404:
             logger.warning(message_for_not_found)
+        else:
+            logger.debug("Kubernetes Exception: " + str(ex))
         if raise_error:
             telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
             raise CLIInternalError(error_message + "\nError Response: " + str(ex.body))
     else:
         if raise_error:
             telemetry.set_exception(exception=ex, fault_type=fault_type, summary=summary)
             raise CLIInternalError(error_message + "\nError: " + str(ex))
```

Comment:
Same comment as above. Probably use ValidationError

Reply:
Same - Do we want to error out here? That branching is not part of raise_error.